rust/src/checkpoints.rs

```diff
@@ -170,6 +170,12 @@ fn parquet_bytes_from_state(state: &DeltaTableState) -> Result<Vec<u8>, Checkpoi
 
     let tombstones = state.unexpired_tombstones();
 
+    // if any, tombstones do not include extended file metadata, we must omit the extended metadata fields from the remove schema
+    // See https://github.com/delta-io/delta/blob/master/PROTOCOL.md#add-file-and-remove-file
+    let use_extended_remove_schema = tombstones
+        .iter()
+        .all(|r| r.extended_file_metadata == Some(true));
```

Comment:
That is to avoid extending parquet schema with null metadata? E.g. so it'll make DBR 8.x to fail I suppose

Reply:
Exactly - this is to avoid writing out the additional fields in the schema and prevent the break in DBR 8.x.